### ISSUE TITLE
2D backwards compatibility

### ISSUE THREAD

**Issue by avaer** [immersive-web/spatial-favicons#3]
I've been experimenting on and off with WebVR/XR favicons for a few years, and I keep coming back to 2D. Getting web developers to make favicon assets is a tall order while the tooling is still immature.

Both sprite extrusion and billboarding seem like a possible on-ramp and zero-cost way to move from the 2D web to the 3D one, but that brings up the question: should we specify how the mapping from 2D to 3D works?

This almost seems more prescient to me than trying to declare a 3D format, for the time being.

**Comment by vyrnix9582046173** [comment]
Found a relevant visual example of favicon sprite extrusions [0]. By billboarding do you mean strictly 2d, and always aligned with the camera? (e.g. not a 3d plane that can be seen from side/behind)


![](https://www.willmcgugan.com/media/uploads/images/favicons.jpg)

[0] https://www.willmcgugan.com/blog/tech/post/turning-website-favicons-in-to-3d-icons/

**Comment by raviramachandra** [comment]
@modulesio  Yes, the choice to creating a 3D Assets lies with the authors. If they do recognize they are in XR world and are willing to provide an appropriate assets it, it will benefit the content's visual representation. But if they choose not to do, how a 2D asset will shown on a platform should be left to the User Agent or the Platform. For Example on Magic Leap devices, the browser just sticks the 2D favicon on the front face of the cube. Sprite Extrusion or Billboarding really can't be dictated on UAs.

**Comment by avaer** [comment]
I’m not sure I understand the backing  argument here.

If displaying assets can’t be dictated on UA’s then this would apply
equally to 2d and 3d content, and perhaps the whole 3D favicons issue
should be out of scope.

I would also posit that becuase the vast majority of sites do not have 3D
favicons, and will not until long after it’s specced, then there is a much
larger win for figuring out the visual representation for existing sites.

And although I don’t think existing implementations should factor in here,
there are others which do extrusion.

**Comment by raviramachandra** [comment]
@modulesio UA's today can choose how to use a favicon in their respective platforms. I did not understand your concerns.
